edx_event_bus_redis/internal/tests/test_consumer.py

```diff
@@ -209,14 +219,64 @@ def raise_exception():
         mock_sleep.assert_not_called()
         self.event_consumer.db.close.assert_called_once_with()  # since shutdown was requested, not because of exception
 
+    @patch('edx_event_bus_redis.internal.consumer.set_custom_attribute', autospec=True)
+    @patch('edx_event_bus_redis.internal.consumer.time.sleep', autospec=True)
+    def test_consume_loop_with_no_events(self, mock_sleep, mock_set_custom_attribute):
+        """
+        Check the loop lifecycle if no events are triggered.
+        """
+        # How the _is_fatal_redis_error() mock will behave on each successive call.
```

Comment:
This mock is confusing. _is_fatal_redis_error does not usually produce any side effects and it feels hacky to use it like this. I think a better test would be to use emit_signals_from_message, whose behavior is less central to the method, and have one call that does not return a message followed by one call that does. That way we can still call `self.event_consumer._shut_down`

Reply:
Thanks for pointing it out and providing a solution, even I was thinking that it seems hacky. Fixed it, please have a look.